src/Summoner/Project.hs

```diff
@@ -0,0 +1,196 @@
+{-# LANGUAGE InstanceSigs #-}
+{-# LANGUAGE QuasiQuotes  #-}
+
+-- | This module introduces functional for project creation.
+
+module Summoner.Project
+       ( Decision (..)
+       , Targets (..)
+       , generateProject
+       ) where
+
+import Control.Monad (when)
+import Data.Aeson (decodeStrict)
+import Data.ByteString.Char8 (pack)
+import Data.List (nub)
+import Data.Semigroup (Semigroup (..))
+import Data.Text (Text)
+import NeatInterpolation (text)
+import System.Info (os)
+import System.Process (readProcess)
+
+import Summoner.Ansi (successMessage, warningMessage)
+import Summoner.Default (currentYear, defaultEmail, defaultGHC, defaultLicense, defaultName,
+                         defaultOwner)
+import Summoner.License (License (..), customizeLicense, githubLicenseQueryNames, licenseNames)
+import Summoner.Process (deleteFile)
+import Summoner.ProjectData (ProjectData (..))
+import Summoner.Question (checkUniqueName, choose, query, queryDef)
+import Summoner.Template (createStackTemplate)
+
+import qualified Data.Text as T
+import qualified Data.Text.IO as T
+
+-- | Used for detecting the user decision during CLI input.
+data Decision = Yes | Nop | Idk
+
+instance Semigroup Decision where
+    (<>) :: Decision -> Decision -> Decision
+    Idk <> x   = x
+    x   <> Idk = x
+    _   <> x   = x
```

Comment:
I think that first test that can be added is semigroup and monoid laws for `Decision`.